cloud_infra_center/ocp_upi/roles/configure-security-groups/tasks/main.yaml

```diff
@@ -0,0 +1,290 @@
+# =================================================================
+# Copyright 2021 https://github.com/openshift/installer
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =================================================================
+
+# Required Python packages:
+#
+# ansible
+# openstackclient
+# openstacksdk
+
+- name: 'Import common yaml'
+  import_tasks: common.yaml
+
+- name: 'Create the master security group'
+  os_security_group:
+    name: "{{ os_sg_master }}"
+
+- name: 'Set master security group tag'
+  command:
+    cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_master }} "
+
+- name: 'Create the worker security group'
+  os_security_group:
+    name: "{{ os_sg_worker }}"
+
+- name: 'Set worker security group tag'
+  command:
+    cmd: "openstack security group set --tag {{ cluster_id_tag }} {{ os_sg_worker }} "
+
+- name: 'Create master-sg rule "ICMP"'
+  os_security_group_rule:
+    security_group: "{{ os_sg_master }}"
+    protocol: icmp
+
+- name: 'Create master-sg rule "machine config server"'
+  os_security_group_rule:
+    security_group: "{{ os_sg_master }}"
+    protocol: tcp
+    remote_ip_prefix: "{{ os_subnet_range }}"
+    port_range_min: 22623
```

Comment:
suggest moving these ports to a defaults variables file or even higher level so they are not hardcoded in the role itself.

Reply:
Good suggestion, and I will consider in the future.